### DISCUSSION TITLE
Reset password comparison with Zod?

### DISCUSSION THREAD

**Discussion by cill-i-am** [colinhacks/zod#1104] [Q&A] [answered]
I'm banging my head against the wall with this, but is there any way to achieve comparing a new password & confirm new password inputs like you would find in a password reset form? 

I can handle it the old fashioned way, but I'm just wondering if i'm missing some clever way using the schema

**Comment by scotttrinh** [comment, marked as answer]
Is something like this what you had in mind?

```ts
const newUserSchema = z
  .object({
    username: z.string().min(2),
    password: z.string().min(8),
    confirm: z.string().min(8),
  })
  .refine(({ password, confirm }) => password === confirm);
```

**Reply by cill-i-am** [reply]
That is exactly what I had in mind. I had no idea refine allowed me to access the rest of the objects fields. Cheers!
